### PR TITLE
Implemented Token whitelisting for CB Pallet 

### LINKED ISSUE #477: Whitelist tokens for Chainbridge

### PR DESCRIPTION
**Pull request by zktony**
Implemented Token whitelisting for CB Pallet. Will close #477

### CODE REVIEW

File: pallets/ocex/src/tests.rs [hunk 1]
```diff
@@ -651,7 +651,7 @@ fn collect_fees() {
 		let snapshot =
 			EnclaveSnapshot::<AccountId32, WithdrawalLimit, AssetsLimit, SnapshotAccLimit> {
 				snapshot_number: 1,
-				merkle_root: mmr_root,
+				snapshot_hash: mmr_root,
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs]: Please update var name as well. Do we need var definition at all here? we can just remove `let mmr_root: H256 = H256::random();` and add `snapshot_hash: H256::random()`

File: pallets/ocex/src/tests.rs [hunk 2]
```diff
@@ -722,7 +722,7 @@ fn test_submit_snapshot_sender_is_not_attested_enclave() {
...
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs]: Same.

File: pallets/ocex/src/tests.rs [hunk 3]
```diff
@@ -744,7 +744,7 @@ fn test_submit_snapshot_snapshot_nonce_error() {
 		let snapshot =
 			EnclaveSnapshot::<AccountId32, WithdrawalLimit, AssetsLimit, SnapshotAccLimit> {
 				snapshot_number: 2,
-				merkle_root: mmr_root,
+				snapshot_hash: mmr_root,
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs]: Same

File: pallets/ocex/src/tests.rs [hunk 4]
```diff
@@ -768,7 +768,7 @@ fn test_submit_snapshot_enclave_signature_verification_failed() {
 		let snapshot =
 			EnclaveSnapshot::<AccountId32, WithdrawalLimit, AssetsLimit, SnapshotAccLimit> {
 				snapshot_number: 1,
-				merkle_root: mmr_root,
+				snapshot_hash: mmr_root,
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs]: Same

File: pallets/ocex/src/tests.rs [hunk 5]
```diff
@@ -791,7 +791,7 @@ fn test_submit_snapshot_bad_origin() {
 		let snapshot =
 			EnclaveSnapshot::<AccountId32, WithdrawalLimit, AssetsLimit, SnapshotAccLimit> {
 				snapshot_number: 0,
-				merkle_root: mmr_root,
+				snapshot_hash: mmr_root,
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs]: Same

File: pallets/ocex/src/tests.rs [hunk 6]
```diff
@@ -833,7 +833,7 @@ fn test_submit_snapshot() {
 		let snapshot =
 			EnclaveSnapshot::<AccountId32, WithdrawalLimit, AssetsLimit, SnapshotAccLimit> {
 				snapshot_number: 1,
-				merkle_root: mmr_root,
+				snapshot_hash: mmr_root,
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs]: Same

File: pallets/ocex/src/tests.rs [hunk 7]
```diff
@@ -960,7 +960,7 @@ fn test_withdrawal() {
...
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs]: Same

File: pallets/ocex/src/tests.rs [hunk 8]
```diff
@@ -1038,7 +1038,7 @@ fn test_onchain_events_overflow() {
 		let snapshot =
 			EnclaveSnapshot::<AccountId32, WithdrawalLimit, AssetsLimit, SnapshotAccLimit> {
 				snapshot_number: 1,
-				merkle_root: hash,
+				snapshot_hash: hash,
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs]: Same

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/ocex/src/tests.rs [hunk 9]
```diff
@@ -1038,7 +1031,7 @@ fn test_onchain_events_overflow() {
 		let snapshot =
 			EnclaveSnapshot::<AccountId32, WithdrawalLimit, AssetsLimit, SnapshotAccLimit> {
 				snapshot_number: 1,
-				merkle_root: hash,
+				snapshot_hash: H256::random(),
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs:1033]: Please remove `hash` var as well.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**